### PR TITLE
#2222 reconnect confirmation page

### PR DESCRIPTION
**Pull request by chrispymm**
This adds a trailing arrow on form routes that don't end in a confirmation page or exit page. 
[Ticket](https://trello.com/c/awSx6IDL/2222-adding-pages-at-the-end-of-the-flow-when-cya-conf-are-not-present-s)

Will show 'Add confirmation page' in the connection menu after a check your answers page
image

Will show 'Reconnect confirmation page' if the confirmation page has become unconnected. This will perform a post request to set the destination of the check your answers page to the confirmation page.
image

### CODE REVIEW

File: app/views/services/edit.html.erb [hunk 1]
```diff
@@ -22,8 +22,9 @@
 <% if ENV['BRANCHING'] == 'enabled' && @detached_flows.present? %>
   <section id="flow-detached" data-component="Expander">
     <h2 class="govuk-heading-s"><%= t('pages.flow.detached') %></h2>
+
     <% @detached_flows.each do |detached_flow| %>
-      <div class="flow-detached-group">
+          <div class="flow-detached-group">
```
Comment (brenetic) [inline app/views/services/edit.html.erb:27]: 🙈

File: app/views/services/_connection_menu.html.erb [hunk 2]
```diff
@@ -68,7 +80,7 @@
           <li data-page-type="singlequestion"
               data-component-type="textarea"><a href="#add-page">Textarea</a></li>
           <li data-page-type="singlequestion"
-              data-component-type="email"><a href="#add-page">Email address</a></li>
+              data-component-type="mail"><a href="#add-page">Email address</a></li>
```
Comment (brenetic) [inline app/views/services/_connection_menu.html.erb:83]: in terms of what the backend knows this component as it is `email`. does the frontend use this differently?

**Comment by brenetic** [comment]
great stuff 🚀 

i would be ok if the last three commits were just one. not a blocker though

**Comment by njseeto** [comment]
LGTM! I just have a similar question re: `data-component-type="mail"` but otherwise it looks great!

File: app/views/services/_connection_menu.html.erb (hunk 2 above)
Comment (chrispymm) [reply app/views/services/_connection_menu.html.erb:83]: No, this looks like I've mistakenly managed to remove the e 🤦‍♂️

**Pull request merged**